src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,248 @@
+    logging.info("Max number of call attempts reached. Returning empty result")
+    return response
+
+
+def get_data_meteo_api(
+    city: str, start_date: str, end_date: str, variable: str, model: str
+) -> pd.DataFrame():
+    """
+    Fetches daily weather data for a specific city and variable from the Meteo API.
+    :param city: The name of the city to fetch the data for. Must be a key in the \
+    COORDINATES dictionary.
+    :param start_date: The start date for the data in the format 'YYYY-MM-DD'.
+    :param end_date: The end date for the data in the format 'YYYY-MM-DD'.
+    :param variable: The weather variable to fetch the data for.
+    :return: A DataFrame containing the fetched data.
+    :raises ValueError: If the specified city is not in the COORDINATES dictionary.
+    """
+    if city not in COORDINATES.keys():
+        raise ValueError("City not available")
+    params = {
+        "latitude": COORDINATES[city]["latitude"],
+        "longitude": COORDINATES[city]["longitude"],
+        "start_date": start_date,
+        "end_date": end_date,
+        "models": model,
+        "daily": variable,
+    }
+    data = call_api(params)
+    if data is None:
+        return None
+    daily_data = data.json()["daily"]
+    # Convert the daily data into a DataFrame
+    df = pd.DataFrame(daily_data)
+    # Add the city name as a column
+    df["city"] = city
+    df["time"] = pd.to_datetime(df["time"])
+    return df
+
+
+def calculate_mean_std(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Calculates the mean and standard deviation for each prefix in the \
+    DataFrame's columns.
+    :param df: The DataFrame to calculate the mean and standard deviation for.
+    :return: A new DataFrame with the mean and standard deviation for each prefix in the 
+    original DataFrame's columns.
+    """
+    df = df.copy()
+    # Extracting prefixes (assuming they are separated by underscores)
+    prefixes = set(col.split("_")[0] for col in df.columns if "_" in col)
+    for prefix in prefixes:
+        relevant_cols = [col for col in df.columns if col.startswith(prefix)]
+        mean_values = df[relevant_cols].mean(axis=1)
+        df[prefix + "_mean"] = mean_values
+        std_values = df[relevant_cols].std(axis=1)
+        df[prefix + "_std"] = std_values
+        df = df.drop(relevant_cols, axis=1)
+    return df
+
+
+def plot_data(df: pd.DataFrame, variable: str) -> None:
+    """
+    Plots the annual mean and dispersion of a specified variable for each city in \
+    the DataFrame.
+    :param df: The DataFrame containing the data to plot.
+    :param variable: The variable to plot.
+    :return: None
+    """
+
+    # Create a mapping from short variable names to full names
+    short_to_full = {v.split("_")[0]: v for v in VARIABLES}
+
+    # Retrieve the full name of the variable
+    full_variable_name = short_to_full.get(variable, variable)
+
+    plt.figure(figsize=(12, 8))
```

Comment:
consider exploring `fig, ax = plt.subplots()` since can give mroe control and be more explicit